### PR TITLE
WIP: Server info v2 #167

### LINKED ISSUE #167: Server info v2
I want to make `/` (root path) as informative and `/status` for getting status of analytics, etc...

Also I like how it is made in elastic
```
{
  "name" : "gCQ5juS",
  "cluster_name" : "elasticsearch",
  "cluster_uuid" : "kNjcFx0aRTW8oq1T4Kr-SQ",
  "version" : {
    "number" : "6.4.0",
    "build_flavor" : "default",
    "build_type" : "zip",
    "build_hash" : "595516e",
    "build_date" : "2018-08-17T23:18:47.308994Z",
    "build_snapshot" : false,
    "lucene_version" : "7.4.0",
    "minimum_wire_compatibility_version" : "5.6.0",
    "minimum_index_compatibility_version" : "5.0.0"
  },
  "tagline" : "You Know, for Search"
}
```

### PR DESCRIPTION
**Pull request by amper43**
Fixes #167

### CODE REVIEW

File: server/src/controllers/analytics_controller.ts [hunk 1]
```diff
@@ -235,6 +235,10 @@ export function isAnalyticReady(): boolean {
   return analyticsService.ready;
 }
 
+export function getZMQConnectionString(): string {
```
Comment (jonyrock) [inline server/src/controllers/analytics_controller.ts]: why not get it from config directly? don`t like this java style much

File: server/src/index.ts [hunk 2]
```diff
@@ -38,12 +38,21 @@ var rootRouter = new Router();
 rootRouter.use('/analyticUnits', anomaliesRouter.routes(), anomaliesRouter.allowedMethods());
 rootRouter.use('/segments', segmentsRouter.routes(), segmentsRouter.allowedMethods());
 //rootRouter.use('/alerts', alertsRouter.routes(), alertsRouter.allowedMethods());
+
+var pjson = require('../../pacakge.json');
```
Comment (jonyrock) [inline server/src/index.ts]: 1. please don`t use `require` for this kind of things. We use webpack which resolves modules. Use another way to load json 
2. name `pjson` is not descriptive. `packageJson` is better
3. use const for this

File: server/src/index.ts [hunk 3]
```diff
@@ -38,12 +38,21 @@ var rootRouter = new Router();
...
+let packageVersion = process.env.npm_package_version ?
+  process.env.npm_package_version : pjson.version;
+
 rootRouter.get('/', async (ctx) => {
   ctx.response.body = {
     server: 'OK',
     analyticsReady: AnalyticsController.isAnalyticReady(),
-    version: process.env.npm_package_version,
-    env: process.env.npm_config_user_agent
+    node_version: process.version,
```
Comment (jonyrock) [inline server/src/index.ts]: Please don`t brage naming conventions

File: server/src/services/analytics_service.ts [hunk 4]
```diff
@@ -14,13 +14,14 @@ export class AnalyticsService {
...
   private _ready: boolean = false;
   private _pingResponded = false;
-  private _zmqConnectionString: string = null;
   private _ipcPath: string = null;
   private _analyticsPinger: NodeJS.Timer = null;
   private _isClosed = false;
   private _productionMode = false;
   private _inDocker = false;
 
+  public zmqConnectionString: string = null;
```
Comment (jonyrock) [inline server/src/services/analytics_service.ts]: Use https://stackoverflow.com/questions/12827266/get-and-set-in-typescript

File: server/src/controllers/analytics_controller.ts (hunk 1 above)
Comment (amper43) [reply server/src/controllers/analytics_controller.ts]: zmqConnectionString builds in runtime and for ipc case you can't get value from config

Comment (jonyrock) [reply server/src/controllers/analytics_controller.ts]: when you compile file `config` file, then it IS the runtime.

File: server/src/services/analytics_service.ts [hunk 5]
```diff
@@ -14,13 +14,14 @@ export class AnalyticsService {
   private _requester: any;
   private _ready: boolean = false;
   private _pingResponded = false;
-  private _zmqConnectionString: string = null;
```
Comment (jonyrock) [inline server/src/services/analytics_service.ts]: so bring it back please

File: server/src/config.ts [hunk 6]
```diff
@@ -43,3 +45,35 @@ function getConfigField(field: string, defaultVal?: any) {
...
+  if(!fs.existsSync(gitHeadFile)) {
+    console.debug(`Can't find git HEAD file ${gitHeadFile}`);
+    return null;
+  }
+  const rev = fs.readFileSync(gitHeadFile).toString();
+  let branchPath = rev.indexOf(':') === -1 ? rev : rev.slice(5, -1);
+  let branch = branchPath.split('/').pop();
+  let commitHash = fs.readFileSync(`${gitRoot}/${branchPath}`).toString().slice(0, -1);
+  return {
+    branch: branch,
```
Comment (rozetko) [inline server/src/config.ts]: You can write just `{ branch, commitHash }` when key name is the same as variable name

File: server/src/config.ts [hunk 7]
```diff
@@ -17,12 +17,16 @@ export const ANALYTIC_UNIT_CACHES_DATABASE_PATCH = path.join(DATA_PATH, 'analyti
...
+export const ZMQ_CONNECTION_STRING = createZMQConnectionString();
 export const ZMQ_IPC_PATH = getConfigField('ZMQ_IPC_PATH', path.join(os.tmpdir(), 'hastic'));
 export const ZMQ_DEV_PORT = getConfigField('ZMQ_DEV_PORT', '8002');
 export const ZMQ_HOST = getConfigField('ZMQ_HOST', '127.0.0.1');
 export const HASTIC_API_KEY = getConfigField('HASTIC_API_KEY');
 export const ANLYTICS_PING_INTERVAL = 500; // ms
+export const PACKAGE_VERSION = getPackageVersion();
+export const GIT_INFO = getGiInfo();
+export const INSIDE_DOCKER = process.env.INSIDE_DOCKER !== undefined;
+export const PRODUCTION_MODE = process.env.NODE_ENV !== 'development'
```
Comment (rozetko) [inline server/src/config.ts]: Forgot `;` ;)

File: server/src/models/analytic_unit_model.ts [hunk 8]
```diff
@@ -80,6 +80,14 @@ export async function findById(id: AnalyticUnitId): Promise<AnalyticUnit> {
   return AnalyticUnit.fromObject(obj);
 }
 
+export async function findAll(id: AnalyticUnitId): Promise<AnalyticUnit[]> {
```
Comment (rozetko) [inline server/src/models/analytic_unit_model.ts]: Why do you need it?

File: server/src/services/analytics_service.ts [hunk 9]
```diff
@@ -214,11 +207,11 @@ export class AnalyticsService {
...
   }
 
-  private static async createIPCAddress(): Promise<{ address: string, file: string }> {
-    let filename = `${process.pid}.ipc`;
+  private static async createIPCAddress(zmqConnectionString: string): Promise<string> {
+    let filename = zmqConnectionString.substring(6); //without 'ipc://'
     let p = path.join(config.ZMQ_IPC_PATH, filename);
     fs.writeFileSync(p, '');
-    return Promise.resolve({ address: 'ipc://' + p, file: p });
+    return Promise.resolve(p);
```
Comment (rozetko) [inline server/src/services/analytics_service.ts]: You don't need `Promise.resolve` if function is `async`
Just `return p;`

File: server/src/services/analytics_service.ts [hunk 10]
```diff
@@ -214,11 +207,11 @@ export class AnalyticsService {
     }, config.ANLYTICS_PING_INTERVAL);
   }
 
-  private static async createIPCAddress(): Promise<{ address: string, file: string }> {
-    let filename = `${process.pid}.ipc`;
+  private static async createIPCAddress(zmqConnectionString: string): Promise<string> {
```
Comment (rozetko) [inline server/src/services/analytics_service.ts]: Why is it `async` if you're writing to file synchronously?

File: server/src/config.ts [hunk 11]
```diff
@@ -77,3 +79,16 @@ function getGiInfo() {
     commitHash: commitHash
   };
 }
+
+function createZMQConnectionString() {
+  let zmq =`tcp://${ZMQ_HOST}:${ZMQ_DEV_PORT}`; //debug mode
```
Comment (jonyrock) [inline server/src/config.ts]: you use this `zmq` only in last the last `return`. I think you can move definition lower.

**Pull request merged**